tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/dd/DeploymentDescriptorServlet.java

```diff
@@ -52,6 +59,120 @@ public class DeploymentDescriptorServlet extends TestServlet {
     @Resource
     private UserTransaction tx;
 
+    // Context Services
+    @Inject
+    private ContextService injectedDefContextSvc;
+
+    @Inject
+    @CustomQualifier1
+    private ContextService injectedContextD;
+
+    @Inject
+    @InvalidQualifier3
+    private ContextService notInjectedContextD;
+    
+    // Managed Executor Services
+    @Inject
+    private ManagedExecutorService injectedDefMES;
+
+    @Inject
+    @CustomQualifier2
+    private ManagedExecutorService injectedMESD;
+
+    @Inject
+    @CustomQualifier2
+    @InvalidQualifier3
+    private ManagedExecutorService notInjectedMESD;
+    
+    // Managed Scheduled Executor Services
+    @Inject
+    private ManagedExecutorService injectedDefMSES;
+
+    @Inject
+    @CustomQualifier1
+    @CustomQualifier2
+    private ManagedExecutorService injectedMSESD;
+
+    @Inject
+    @CustomQualifier1
+    private ManagedExecutorService notInjectedMSESD;
+    
+    // Managed Thread Factory
+    @Inject
+    private ManagedThreadFactory injectedDefMTF;
+
+    @Resource(lookup = "java:app/concurrent/ThreadFactoryD")
+    private ManagedThreadFactory resourceMTFD;
+
+    @Inject
+    @CustomQualifier1
+    private ManagedThreadFactory notInjectedMTFD;
+
+    public void testDeploymentDescriptorDefinesQualifiers() throws Throwable {
+        ContextService lookupDefContextSvc = InitialContext.doLookup("java:comp/DefaultContextService");
+        ContextService lookupContextD = InitialContext.doLookup("java:global/concurrent/ContextD");
+
+        assertAll("Context Service Tests",
+                () -> assertNotNull(injectedDefContextSvc,
+                        "Default contextService was not injected when no qualifiers were present."),
+                () -> assertEquals(lookupDefContextSvc, injectedDefContextSvc,
```

Comment:
I don't think we can assert equality of looked up instances because I don't know of any specification requirement for the same instance to be returned.  A valid implementation could return a different instance as long as it behaves correctly.